mmpose/datasets/pipelines/mesh_transform.py

```diff
@@ -0,0 +1,384 @@
+                                         get_affine_transform)
+from mmpose.datasets.registry import PIPELINES
+
+
+def _flip_smpl_pose(pose):
+    """Flip SMPL pose parameters horizontally.
+
+    Args:
+        pose (np.ndarray([72])): SMPL pose parameters
+
+    Returns:
+        pose_flipped
+    """
+
+    flippedParts = [
+        0, 1, 2, 6, 7, 8, 3, 4, 5, 9, 10, 11, 15, 16, 17, 12, 13, 14, 18, 19,
+        20, 24, 25, 26, 21, 22, 23, 27, 28, 29, 33, 34, 35, 30, 31, 32, 36, 37,
+        38, 42, 43, 44, 39, 40, 41, 45, 46, 47, 51, 52, 53, 48, 49, 50, 57, 58,
+        59, 54, 55, 56, 63, 64, 65, 60, 61, 62, 69, 70, 71, 66, 67, 68
+    ]
+    pose_flipped = pose[flippedParts]
+    # Negate the second and the third dimension of the axis-angle
+    pose_flipped[1::3] = -pose_flipped[1::3]
+    pose_flipped[2::3] = -pose_flipped[2::3]
+    return pose_flipped
+
+
+def _flip_iuv(iuv, uv_type='BF'):
+    """Flip IUV image horizontally.
+
+    Notes:
+        IUV image height: H
+        IUV image width: W
+
+    Args:
+        iuv np.ndarray([H, W, 3]): IUV image
+        uv_type (str): The type of the UV map.
+            Meaningful values:
+                'DP': The UV map used in DensePose project.
+                'SMPL': The default UV map of SMPL model.
+                'BF': The UV map used in DecoMR project.
+            Default: 'BF'
+
+    Returns:
+        iuv_flipped np.ndarray([H, W, 3]): Flipped IUV image
+    """
+    assert uv_type in ['DP', 'SMPL', 'BF']
+    if uv_type == 'BF':
+        iuv_flipped = iuv[:, ::-1, :]
+        iuv_flipped[:, :, 1] = 255 - iuv_flipped[:, :, 1]
+    else:
+        # The flip of other UV map is complex, not finished yet.
+        raise NotImplementedError(
+            f'The flip of {uv_type} UV map is not implemented yet.')
+
+    return iuv_flipped
+
+
+def _rotate_joints_3d(joints_3d, rot):
+    """Rotate the 3D joints in the local coordinates.
+
+    Notes:
+        Joints number: K
+
+    Args:
+        joints_3d (np.ndarray([K, 3])): Coordinates of keypoints.
+        rot (float): Rotation angle (degree).
+
+    Returns:
+        joints_3d_rotated
+    """
+    # in-plane rotation
+    rot_mat = np.eye(3)
+    if rot != 0:
+        rot_rad = -rot * np.pi / 180
+        sn, cs = np.sin(rot_rad), np.cos(rot_rad)
+        rot_mat[0, :2] = [cs, -sn]
+        rot_mat[1, :2] = [sn, cs]
+
+    joints_3d_rotated = np.einsum('ij,kj->ki', rot_mat, joints_3d)
+    joints_3d_rotated = joints_3d_rotated.astype('float32')
+    return joints_3d_rotated
+
+
+def _rotate_smpl_pose(pose, rot):
+    """Rotate SMPL pose parameters. SMPL (https://smpl.is.tue.mpg.de/) is a 3D
+    human model.
+
+    Args:
+        pose (np.ndarray([72])): SMPL pose parameters
+        rot (float): Rotation angle (degree).
+
+    Returns:
+        pose_rotated
+    """
+    pose_rotated = pose
+    if rot != 0:
+        R = np.array([[np.cos(np.deg2rad(-rot)), -np.sin(np.deg2rad(-rot)), 0],
```

Comment:
how about define a util function named `_construct_rotation_matrix`, and use it here, and also in function `_rotate_joints_3d`.

it is just odd to see that a same thing is doing very differently.

Reply:
ok, I will do this.